### PR TITLE
Pisp/305 auth e2e tests

### PR DESCRIPTION
**Pull request by eoln**
finalize feature POST /authorizations:
- improve unit tests
- adjust PUT /authorizations payload schema

### CODE REVIEW

File: src/interface/TransactionRequestsService-swagger.yaml [hunk 1]
```diff
@@ -374,7 +374,8 @@ paths:
           The Accept header field indicates the version of the API the client
           would like the server to use.
         in: header
-        required: true
+        # TODO: fix mojaloop_simulator getSignedChallenge response first
```
Comment (eoln) [inline src/interface/TransactionRequestsService-swagger.yaml:377]: @sridharvoruganti there is a need to add `accept` header in the mocked response in mojaloop_simulator

File: src/interface/swagger.json [hunk 2]
```diff
@@ -950,9 +950,33 @@
         "AuthenticationValue": {
             "title": "AuthenticationValue",
             "description": "Contains the authentication value. The format depends on the authentication type used in the AuthenticationInfo complex type.",
-            "pattern": "^\\d{3,10}$|^\\S{1,64}$",
             "type": "string"
```
Comment (eoln) [inline src/interface/swagger.json:953]: unfortunately swagger 2.0 doesn't support oneOf, so auto mocking will not work for U2F case

File: src/domain/authorizations/authorizations.js [hunk 3]
```diff
@@ -78,8 +78,15 @@ async function forwardAuthorizationMessage (headers, transactionRequestId, paylo
       )
     }
 
-    const query = method === Enum.Http.RestMethods.GET ? '?' + (new URLSearchParams(payload).toString()) : ''
-    const fullUrl = `${endpoint}/authorizations/${transactionRequestId}${query}`
+    let fullUrl
+    if (method === Enum.Http.RestMethods.GET) {
```
Comment (lewisdaly) [inline src/domain/authorizations/authorizations.js:82]: Minor nitpick: could this be changed to `switch` instead? I generally find that much more readable

Comment (eoln) [reply src/domain/authorizations/authorizations.js]: @rmothilal & @lewisdaly - Aye Aye - fixed

File: src/interface/swagger.json (hunk 2 above)
Comment (rmothilal) [reply src/interface/swagger.json:953]: @eoln are you not editing the https://github.com/mojaloop/transaction-requests-service/blob/pisp/master/src/interface/TransactionRequestsService-swagger.yaml also?

Comment (eoln) [reply src/interface/swagger.json:953]: @rmothilal I edited it: https://github.com/mojaloop/transaction-requests-service/pull/47/files#diff-d5a56fe5aa69dae0c6905cb159ee9b4a

**Pull request merged**